airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -219,6 +219,90 @@ def _validate_empty_streams(self, records, configured_catalog, allowed_empty_str
         streams_without_records = streams_without_records - allowed_empty_streams
         assert not streams_without_records, f"All streams should return some records, streams without records: {streams_without_records}"
 
+    def _unpack_record_with_schema(self, record: Dict, top_level_field: str, inner_props: dict):
```

Comment:
does it work with oneOf   ?  I would suggest to use get_expected_schema_structure and get_object_structure functions from json_schema_helper.py. In this case code would be like :+1: 
```python
expected_pathes = get_expected_schema_structure(schema, annotate_one_of = True)
expected_pathes = set(flatten(expected_pathes) # function doesnt exist)
for record in records:
    record_pathes = set(get_object_structure(record.data))
    expected_pathes -=  record_pathes
assert expected_pathes
```